src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs

```diff
@@ -78,6 +81,29 @@ public static void UpdateRunSettingsWithUserProvidedSwitches(XPathNavigator runS
         }
 
         /// <summary>
+        /// Updates the <c>RunConfiguration.DesignMode</c> value for a run settings. Doesn't do anything if the value is already set.
+        /// </summary>
+        /// <param name="runSettingsNavigator">Navigator for runsettings xml</param>
+        /// <param name="designModeValue">Value to set</param>
+        public static void UpdateDesignMode(XPathNavigator runSettingsNavigator, bool designModeValue)
+        {
+            // Navigator should be at Root of runsettings xml, attempt to move to /RunSettings/RunConfiguration
+            if (!runSettingsNavigator.MoveToChild(RunSettingsNodeName, string.Empty) ||
+!runSettingsNavigator.MoveToChild(RunConfigurationNodeName, string.Empty))
```

Comment:
should we add a RunConfiguration node if it isn't available? Don't think we can expect the user to have this always. Or does the platform already fill in a default RunConfiguration node by now?